stellargraph/connector/neo4j/graph.py

```diff
@@ -138,6 +139,50 @@ def to_adjacency_matrix(self, node_ids):
     def is_directed(self):
         return self._is_directed
 
+    def check_graph_for_ml(self):
+        """
+        Checks if all properties required for machine learning training/inference are set up.
+        An error will be raised if the graph is not correctly setup.
+        """
+        num_nodes_with_feats_query = f"""
+            MATCH(n)
+            WHERE "features" in keys(n)
+            return COUNT(*)
+        """
+        result = self.graph_db.run(num_nodes_with_feats_query)
+        num_nodes_with_feats = result.data()[0]["COUNT(*)"]
+        if num_nodes_with_feats == 0:
+            raise RuntimeError(
+                "This StellarGraph has no numeric feature attributes for nodes"
+                "Node features are required for machine learning"
+            )
+
+    def unique_node_type(self, error_message=None):
+        """
+        Return the unique node type, for a homogeneous-node graph.
+
+        Args:
+            error_message (str, optional): a custom message to use for the exception; this can use
+                the ``%(found)s`` placeholder to insert the real sequence of node types.
+
+        Returns:
+            If this graph has only one node type, this returns that node type, otherwise it raises a
+            ``ValueError`` exception.
+        """
+        node_type_query = f"""MATCH (n) RETURN distinct labels(n)"""
+        result = self.graph_db.run(node_type_query)
+        all_types = result.data()[0]["labels(n)"]
+        if len(all_types) == 1:
+            return all_types[0]
```

Comment:
Is it a problem that the `labels` returns a list of individual labels (because Neo4j allows multiple labels per node), and so this is likely to be returning a list?

Reply:
I'm not so sure that checking for the number of labels in the graph is the best way to check for node types since each node can have multiple labels, and so we don't have a good 1-to-1 mapping from Neo4j labels to stellargraph types.. 

I went with a different approach in https://github.com/stellargraph/stellargraph/pull/1595/files where we just assume that every node is of default type.

We would want to support heterogeneous graphs with Neo4j in the future, but might need a bit more logic to do the mapping (unique set of labels -> type?). So for now, I'd prefer not to raise an error here based on the existence of labels - thoughts @huonw ?

Reply:
Sounds reasonable for now. Maybe we should have an issue covering Neo4j <-> SG types?

Reply:
I've filed https://github.com/stellargraph/stellargraph/issues/1625